File: java/simpleserver/src/main/java/com/rc/simpleserver/engine/endpoint/DefaultDispatcher.java
package com.rc.simpleserver.engine.endpoint;

import org.springframework.stereotype.Component;

@Component
public class DefaultDispatcher implements IEndPointDispatcher {
    @Override
    public String dispatch() {
        return "{\n" +
                "    \"msg\": \"default implementation\"\n" +
                "}\n";
    }

    @Override
    public String dispatchGet() {
        return dispatch();
    }
}

File: kotlin/simpleserver/src/main/kotlin/com/rc/simpleserver/engine/endpoint/DefaultDispatcher.kt
package com.rc.simpleserver.engine.endpoint

class DefaultDispatcher : EndPointDispatcher {
    override fun dispatch(): String {
        return """{"msg": "default implementation"}"""
    }

    override fun dispatchGet(): String {
        return dispatch()
    }

    override fun dispatchPost(): String {
        return dispatch()
    }
}
